### PR TITLE
Minor secrets refactor

### PR DESCRIPTION
**Pull request by cicdw**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
This PR does a few things:
- renames our secret tasks for clarity
- allows for new secret names to be passed at runtime
- removes `env_var` as a distinct option from `EnvVarSecret`

## Why is this PR important?
This PR was originally motivated by our work on Results (additional PR for `SecretResult` incoming), and I went ahead and changed the names as well as this has been a common source of confusion.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ These changes are available in the [master branch](https://github.com/PrefectHQ/
 
 ### Breaking Changes
 
-- None
+- Remove `env_var` initialization from `EnvVarSecret` in favor of `name` - [#2302](https://github.com/PrefectHQ/prefect/pull/2302)
```
Comment (cicdw) [inline CHANGELOG.md:32]: Note that I don't believe anyone is actually using this task, as it's more of a nice example than a truly useful thing.

File: src/prefect/tasks/secrets/base.py [hunk 2]
```diff
@@ -3,16 +3,37 @@
 from prefect.client.secrets import Secret as _Secret
 from prefect.core.task import Task
 from prefect.engine.result_handlers import SecretResultHandler
+from prefect.utilities.tasks import defaults_from_attrs
 
 
-class Secret(Task):
```
Comment (joshmeek) [inline src/prefect/tasks/secrets/base.py:8]: To ensure the non-breaking of this we could keep the `Secret` name w/ a deprecation warning and formally drop in 0.11.0

Comment (cicdw) [reply src/prefect/tasks/secrets/base.py:8]: FYI I do keep the `Secret` name at the bottom of the file; I need to think about how I can raise a deprecation warning if it's used though 🧐

Comment (cicdw) [reply src/prefect/tasks/secrets/base.py:8]: ah I know, will update

Comment (cicdw) [reply src/prefect/tasks/secrets/base.py:8]: @joshmeek updated

**Pull request merged**